src/griddyn/relays/controlRelay.h

```diff
@@ -93,15 +93,15 @@ class controlRelay : public Relay
 protected:
   virtual void actionTaken (index_t ActionNum, index_t conditionNum, change_code actionReturn, coreTime actionTime) override;
 
-  virtual void receiveMessage (std::uint64_t sourceID, std::shared_ptr<commMessage> message) override;
+  virtual void receiveMessage (std::uint64_t sourceID, std::shared_ptr<commMessage> const& message) override;
```

Comment:
This is a virtual function we would need to make sure all uses are changed likewise and are amenable to the change,  which given this can be called in a multithread situation I am not sure I am comfortable passing around a reference to a shared pointer.

Reply:
I changed all uses. Whatever calls receiveMessage will have ownership of message, so there should always be at least one reference, right? I'm not sure how it could be destructed even with threads.